src/hdmf/common/table.py

```diff
@@ -64,22 +48,26 @@ def add_row(self, **kwargs):
 
 
 @register_class('VectorIndex')
-class VectorIndex(Index):
+class VectorIndex(VectorData):
     """
     When paired with a VectorData, this allows for storing arrays of varying
     length in a single cell of the DynamicTable by indexing into this VectorData.
     The first vector is at VectorData[0:VectorIndex(0)+1]. The second vector is at
     VectorData[VectorIndex(0)+1:VectorIndex(1)+1], and so on.
     """
 
+    __fields__ = ("target",)
+
     @docval({'name': 'name', 'type': str, 'doc': 'the name of this VectorIndex'},
             {'name': 'data', 'type': ('array_data', 'data'),
              'doc': 'a 1D dataset containing indexes that apply to VectorData object'},
             {'name': 'target', 'type': VectorData,
              'doc': 'the target dataset that this index applies to'})
     def __init__(self, **kwargs):
+        target = getargs('target', kwargs)
+        kwargs['description'] = "Index for VectorData '%s'" % target.name
```

Comment:
Note that `description` is set by the API since it is required for `VectorData` but did not used to exist in `VectorIndex`